src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -238,10 +227,108 @@ object FunctionRegistry {
           v = args(2).run(ec)();
 
           _ = (if (v == null)
-            fatal(s"Argument evaluated to missing in call to aggregator $name"));
+            fatal(s"Argument evaluated to missing in call to aggregator $name"))
+
+        ) yield aggregator.ctor(g, v.asInstanceOf[v])
+
+      case aggregator: GroupByAggregatorFun[t] =>
+        val Lambda(_, keyParam, keyBody) = args(1)
+        val TFunction(Seq(keyParamType), _) = argTypes(1)
+
+        val Lambda(_, downstreamParam, downstreamBody) = args(2)
+        val TFunction(Seq(downstreamParamType), _) = argTypes(2)
```

Comment:
You can put a type constraint `downstreamParamType: TAggregable` and remove the casts below.